lib/galaxy/tools/__init__.py

```diff
@@ -664,6 +664,8 @@ def _view(self):
 
     @property
     def version_object(self):
+        if not self.version:
+            raise Exception("Must call parse() first")
```

Comment:
`parse()` is called by `__init__()` ?

Reply:
Agreed, but without this we get https://app.circleci.com/pipelines/github/galaxyproject/galaxy/19099/workflows/84a60b86-6e6b-46f3-909b-62844104652b/jobs/156302?invite=true#step-104-2387

> `galaxy/tools/__init__.py:667: error: Argument 1 to "parse" has incompatible type "Optional[Any]"; expected "str"  [arg-type]`